kwnlp_preprocessor/tests/test-end-to-end-no-download.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2021-present Kensho Technologies, LLC.
+import os.path
+import filecmp
+import logging
+
+from kwnlp_preprocessor import argconfig
+from kwnlp_preprocessor import task_03p1_create_kwnlp_pagecounts
+from kwnlp_preprocessor import task_03p2_convert_sql_to_csv
+from kwnlp_preprocessor import task_06p1_create_kwnlp_page_props
+from kwnlp_preprocessor import task_06p2_create_kwnlp_redirect_it2
+from kwnlp_preprocessor import task_09p1_create_kwnlp_ultimate_redirect
+from kwnlp_preprocessor import task_12p1_create_kwnlp_title_mapper
+from kwnlp_preprocessor import task_15p1_split_and_compress_wikidata
+from kwnlp_preprocessor import task_18p1_filter_wikidata_dump
+from kwnlp_preprocessor import task_21p1_gather_wikidata_chunks
+from kwnlp_preprocessor import task_24p1_create_kwnlp_article_pre
+from kwnlp_preprocessor import task_27p1_parse_wikitext
+from kwnlp_preprocessor import task_30p1_post_process_link_chunks
+from kwnlp_preprocessor import task_33p1_collect_post_processed_link_data
+from kwnlp_preprocessor import task_36p1_collect_template_data
+from kwnlp_preprocessor import task_36p2_collect_length_data
+from kwnlp_preprocessor import task_39p1_create_kwnlp_article
+from kwnlp_preprocessor import task_42p1_collect_section_names
+import unittest
+import shutil
+from tempfile import TemporaryDirectory
+import os
+
+logger = logging.getLogger(__name__)
+
+
+def are_dir_trees_equal(output: str, truth: str) -> bool:
+    """Compare two directories recursively.
+
+    Files in each directory are assumed to be equal if their names and contents are equal.
+    This is a modified version found from https://stackoverflow.com/a/6681395, it only cares that all the files in the truth are in the output, but not the other way around.
+    The output will have extra files because there is intermediate data we dont
+    care about comparing. The intermediate data are implementation details.
+
+    @param output: Outputs from processing
+    @param truth: Ground truth outputs
+
+    @return: True if the directory trees are the same and
+        there were no errors while accessing the directories or files,
+        False otherwise.
+    """
+    dirs_cmp = filecmp.dircmp(output, truth)
+    # If the truth data has files the output doesnt, we fail
+    # funny_files: Files which are in both a and b, but could not be compared.
+    if len(dirs_cmp.right_only) > 0 or len(dirs_cmp.funny_files) > 0:
+        return False
+    (_, mismatch, errors) = filecmp.cmpfiles(output, truth, dirs_cmp.common_files, shallow=False)
```

Comment:
nit: I don't think you need the parens around `_, mismatch, errors`.

Reply:
I might have copy and pasted this 👀  good catch.